src/publisher/tests/test_fragment_logic.py

```diff
@@ -115,20 +117,66 @@ def test_valid_merge_updates_article_version_fields(self):
         self.assertTrue(av.article_json_v1)
         self.assertTrue(av.article_json_v1_snippet)
 
-    def test_merge_sets_status_date_correctly(self):
-        # setUp inserts article snippet that should be valid
+    def test_merge_sets_status_date_correctly_poa_v1(self):
+        "statusDate for a v1 poa is correctly set: earliest POA if POA, earliest VOR if VOR"
+        # poa, published 2016-08-16T00:00:00Z
         av = self.freshen(self.av)
-        placeholders = {'statusDate': '2001-01-01T00:00:00Z', }
-        logic.add(self.msid, 'foo', placeholders)
-
-        self.assertTrue(logic.merge_if_valid(self.av))
+        expected = '2016-08-16T00:00:00Z'
+        self.assertEqual(expected, av.article_json_v1['statusDate'])
 
+    def test_merge_sets_status_date_correctly_poa_v2(self):
+        "statusDate for a v2 poa is correctly set: earliest POA if POA, earliest VOR if VOR"
         av = self.freshen(self.av)
-        self.assertTrue(av.article_json_v1)
-        self.assertTrue(av.article_json_v1_snippet)
+        expected = '2016-08-16T00:00:00Z'
+        self.assertEqual(expected, av.article_json_v1['statusDate'])
+
+        # load v2 as a poa
+        fixture = join(self.fixture_dir, 'ajson', 'elife-16695-v2.xml.json')
+        data = json.load(open(fixture, 'r'))
+        data['status'] = models.POA
+        data['published'] = '2016-08-17T00:00:00Z' # v2 POA published a day later
+        _, _, av2 = ajson_ingestor.ingest_publish(data)
+
+        # nothing should have changed
+        av2 = self.freshen(av)
+        self.assertEqual(expected, av2.article_json_v1['statusDate'])
+
+    def test_merge_sets_status_date_correctly_vor_v2(self):
+        "statusDate for a v2 vor is correctly set: earliest POA if POA, earliest VOR if VOR"
+        # load v2 vor
+        fixture = join(self.fixture_dir, 'ajson', 'elife-16695-v2.xml.json')
```

Comment:
I don't mind repeated code in test cases - I'd rather it were all painfully explicit and stupid, rather than clever or optimised.